server/models/BlogPost.js

```diff
@@ -0,0 +1,15 @@
+import mongoose from 'mongoose';
+const { Schema } = mongoose;
```

Comment:
are you not missing to import model?

`const { Schema, model } = require("mongoose");`